### ISSUE TITLE
NullReference Exception under Android API 25

### ISSUE THREAD

**Issue by j-sauer** [halkar/xamarin-range-slider#117]
Hi,

the current version does not work with Android. I always get a NullReferenceException.
Target API is 25.
Test project can be found here: [RangeSliderTest](https://github.com/j-sauer/RangeSliderTest)

**Comment by halkar** [comment]
Thanks @j-sauer 
Apparently, range slider ios package is not compatible with new Android projects. I will try to fix it asap.

**Comment by timdinhdotcom** [comment]
I think the `NullReferenceException` is due to `ThumbImage` being null `RangeSliderControl.cs` line 328
```
_thumbHalfWidth = 0.5f * ThumbImage.Width;
```

`ThumbImage` was set by the line
```
ThumbImage = BitmapFactory.DecodeResource(Resources, thumbNormal);
```

**Comment by charlesroddie** [comment]
@timdinhdotcom Any ideas why android isn't loading the image?

**Comment by charlesroddie** [comment]
@j-sauer is finding weird issues. seek_thumb_normal.png and seek_thumb_pressed.png don't load. seek_thumb_disabled does seem to load but looks like a faint rectangle rather than a circle.

**Comment by timdinhdotcom** [comment]
Here is my work around

Add this method
```
private Bitmap BitmapFromResourceId(int id)
{
    using (var stream = Resources.OpenRawResource(id))
    {
        return BitmapFactory.DecodeStream(stream);
    }
}
```

change this block 
```
            if (ThumbImage == null)
            {
                ThumbImage = BitmapFactory.DecodeResource(Resources, thumbNormal);
            }
            if (ThumbPressedImage == null)
            {
                ThumbPressedImage = BitmapFactory.DecodeResource(Resources, thumbPressed);
            }
            if (ThumbDisabledImage == null)
            {
                ThumbDisabledImage = BitmapFactory.DecodeResource(Resources, thumbDisabled);
            }

```
to 
```
            if (ThumbImage == null)
            {
                ThumbImage = BitmapFromResourceId(thumbNormal);
            }
            if (ThumbPressedImage == null)
            {
                ThumbPressedImage = BitmapFromResourceId(thumbPressed);
            }
            if (ThumbDisabledImage == null)
            {
                ThumbDisabledImage = BitmapFromResourceId(thumbDisabled);
            }

```

**Comment by charlesroddie** [comment]
@timdinhdotcom OK great. @j-sauer and I are working on doing this with circle drawables, bypassing resources. If our approach is going well we can take that one (which is resolution independent), otherwise we can use yours. We can do the PR.

**Comment by halkar** [comment]
Thanks @timdinhdotcom. But your changes don't fix @j-sauer issue.

**Comment by halkar** [comment]
Thanks @charlesroddie How did you fix it? I've been trying to fix it for a while but didn't fix it.

**Comment by charlesroddie** [comment]
We are trying to bypass using resources by putting drawing circles in code instead of using bitmaps. (Could use xml too but code to be completely safe.) All of the relevant resources (seek_thumb_disabled, _normal, and _pressed) are just two or three circles. Pretty sure that will work.

**Comment by halkar** [comment]
Fixed in 0.7.30

Thanks a lot @charlesroddie
